### PR TITLE
[ios] Adds pointForCoordinate/coordinateForPoint to MGLMapSnapshotOverlay

### PR DESCRIPTION
**Pull request by julianrex**
Fixes https://github.com/mapbox/mapbox-gl-native/issues/15695, by adding the same point & coordinate methods as `MGLMapSnapshot`

EDIT: Here's a test from iOS, drawing a line from coordinates for London<->Paris

Screen Shot 2019-10-02 at 10 34 45 AM

macOS:
streets-v10

### CODE REVIEW

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 
 ### Other changes
 
+* Added `-[MGLMapSnapshotOverlay coordinateForPoint:]` and `-[MGLMapSnapshotOverlay pointForCoordinate:]` that mirrors those of `MGLMapSnapshot`. ([#15746](https://github.com/mapbox/mapbox-gl-native/pull/15746))
```
Comment (captainbarbosa) [inline platform/ios/CHANGELOG.md]: I would also describe what these methods are used for. Perhaps something like:

>Added `-[MGLMapSnapshotOverlay coordinateForPoint:]` and `-[MGLMapSnapshotOverlay pointForCoordinate:]` to convert between context and map coordinates, mirroring those of `MGLMapSnapshot`.

File: platform/ios/Integration Tests/Snapshotter Tests/MGLMapSnapshotterTest.m [hunk 2]
```diff
@@ -462,4 +478,60 @@ - (void)testSnapshotWithOverlayHandlerSuccess🔒 {
     [self waitForExpectations:@[expectation] timeout:10.0];
 }
 
+- (void)testSnapshotCoordinatesWithOverlayHandler🔒 {
```
Comment (captainbarbosa) [inline platform/ios/Integration Tests/Snapshotter Tests/MGLMapSnapshotterTest.m]: Not specific to this PR, but I would love to see a reference as to that 🔒means here.

File: platform/ios/CHANGELOG.md (hunk 1 above)
Comment (julianrex) [reply platform/ios/CHANGELOG.md]: Like it! Will change.

File: platform/ios/Integration Tests/Snapshotter Tests/MGLMapSnapshotterTest.m (hunk 2 above)
Comment (julianrex) [reply platform/ios/Integration Tests/Snapshotter Tests/MGLMapSnapshotterTest.m]: 👍

File: platform/darwin/src/MGLMapSnapshotter.mm [hunk 3]
```diff
@@ -33,23 +35,61 @@
...
+{
+    mbgl::ScreenCoordinate sc = _pointForFn(MGLLatLngFromLocationCoordinate2D(coordinate));
+    CGFloat height = ((CGFloat)CGBitmapContextGetHeight(self.context))/self.scale;
+    return NSMakePoint(sc.x, height - sc.y);
+}
+
+- (CLLocationCoordinate2D)coordinateForPoint:(NSPoint)point
+{
+    CGFloat height = ((CGFloat)CGBitmapContextGetHeight(self.context))/self.scale;
+    auto screenCoord = mbgl::ScreenCoordinate(point.x, height - point.y);
```
Comment (1ec5) [inline platform/darwin/src/MGLMapSnapshotter.mm]: Thanks for accounting for the difference in coordinate system origin on macOS! ❤️

**Comment by emtjoshhart** [comment]
How could one implement a polygon on top of a snapshot overlay?

**Pull request merged**